### PR TITLE
Replace `Map` with `Array` for `unparentedViews`

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What
Drop last use of `Map` in the stateful clients' state.
Using an `Array` for the `unparentedViews` doesn't have major performance implications because the total number of entries is limited by the number of video devices available locally and because the list is not iterated over unconditionally (there is no base selector on this list).

# Why
Play well with Redux.

# How Tested
* `rush build`
* `rush test`
* Local storybook, toggle preview on configuration screen.

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [x] This change causes current functionality to break.

### CODE REVIEW

File: packages/calling-stateful-client/src/CallContext.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class CallContext {
         cameras: [],
         microphones: [],
         speakers: [],
-        unparentedViews: new Map<LocalVideoStreamState, LocalVideoStreamState>()
```
Comment (JamesBurnside) [inline packages/calling-stateful-client/src/CallContext.ts:55]: this used to be a map of an object.. to itself?

Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallContext.ts:55]: (where key and value the same here?)

Comment (JamesBurnside) [reply packages/calling-stateful-client/src/CallContext.ts:55]: never mind read the comment where this is defined

**Review by JamesBurnside — APPROVED** [review]

**Review by anjulgarg — APPROVED** [review]

File: packages/calling-stateful-client/src/CallContext.ts (hunk 1 above)
Comment (prprabhu-ms) [reply packages/calling-stateful-client/src/CallContext.ts:55]: Yeah, this was a very weird map. I had to double and triple check that we didn't actually depend on the keys being referentially equal across some API calls.

**Pull request merged**